cfme/tests/infrastructure/test_vm_delete.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+
+from cfme import test_requirements
+from cfme.infrastructure.provider.rhevm import RHEVMProvider
+from cfme.utils.generators import random_vm_name
+from cfme.utils.log_validator import LogValidator
+from cfme.utils.wait import TimedOutError
+from cfme.utils.wait import wait_for
+
+pytestmark = [
+    pytest.mark.provider([RHEVMProvider], scope="module"),
+    pytest.mark.usefixtures('setup_provider'),
+    test_requirements.rhev
+]
+
+
+@pytest.fixture(scope="module")
+def vm_test(provider):
+    collection = provider.appliance.provider_based_collection(provider)
+    vm_name = random_vm_name(context="del-test")
+    vm = collection.instantiate(vm_name, provider)
+    vm.create_on_provider(allow_skip="default", find_in_cfme=True)
+    vm.wait_to_appear(timeout=900, load_details=False)
+    yield vm
+
+    if vm.exists:
+        vm.cleanup_on_provider()
+
+
+@pytest.mark.rhv2
+@pytest.mark.meta(automates=[1592430])
+def test_delete_vm_on_provider_side(vm_test, provider):
+    """ Delete VM on the provider side and refresh relationships in CFME
+
+    Polarion:
+        assignee: anikifor
+        initialEstimate: 1/4h
+        casecomponent: Infra
+
+    Bugzilla:
+        1592430
+    """
+    logs = LogValidator("/var/www/miq/vmdb/log/evm.log", failure_patterns=[".*ERROR.*"])
+    logs.start_monitoring()
+    vm_test.cleanup_on_provider()
```

Comment:
**Question** Do you think it is worth verifying that the vm was deleted?

Reply:
@jawatts don't think so... 
`cleanup` method calls `delete` and it's the wait_for for that
```
    def delete(self):
        """
        Removes the entity on the provider
        """
        self.ensure_state(VmState.STOPPED)
        self.logger.debug(' Deleting RHEV VM %s/%s', self.name, self.uuid)

        self.api.remove()

        wait_for(
            lambda: not self.exists,
            message="wait for RHEV VM '{}' deleted".format(self.uuid),
            num_sec=300
        )
        return True
```

Reply:
Yes! I thought so!